### PR TITLE
Added support for RHEL7 and created a fstab:params class

### PR DESCRIPTION
**Pull request by bschonec**
Moved some code out into fstab::params class to consolidate pseudo-duplicate code.

### CODE REVIEW

File: manifests/params.pp [hunk 1]
```diff
@@ -0,0 +1,32 @@
+# Standard parameters for the fstab class
+
+class fstab::params {
+
+case $::osfamily {
+  
+  'Debian': { 
+
+		$nfs_client_package = ['nfs-common',]
```
Comment (chrisjohnson1988) [inline manifests/params.pp]: Mixing tabs and spaces

**Comment by chrisjohnson1988** [comment]
Hey @bschonec 

Sorry for the delay in responding to this. I like the general idea of these changes.

I have a few issues with the **fstab::params** class which would need to be resolved before merging.

This project uses _spaces_ over _tabs_ for indentation, I would like to see this file updated to using spaces.

Also the indentation of the file seems a little off.

Many Thanks,
Chris

**Comment by bschonec** [comment]
I have [hopefully] fixed the tabs and spaces issues.  Forgive me as I'm
still quite green at all of this.

**Comment by BasLangenberg** [comment]
Hi, I'm using this for RHEL 6 boxes as well and we are in the progress of upgrading to RHEL7. I would really like to see this PR merged!

Is there something I can do to help out?

**Comment by bschonec** [comment]
As I have zero knowledge on how to use travis, any help to ME would certainly help this merge request.  ;-)

**Comment by chrisjohnson1988** [comment]
Hey Guys,

Sorry about the long delays in dealing with this. It looks as if there were a few conflicts in place which I have just resolved.

@bschonec As you mentioned travis, if you feel like adding automated testing to this repository that would be very much appreciated. I think it is fair to say that our other puppet modules have some pretty good examples of travis run tests in place.

**Comment by bschonec** [comment]
@chrisjohnson1988, although I have tried many times to build .spec files it escapes me as to why and how this is done.  I'm a linux admin, not a developer by trade so I've had nobody to explain the "why" part.   I'm still trying, though!  ;-)

**Comment by chrisjohnson1988** [comment]
The _why_ part is simply to remove as many silly mistakes as possible. It also helps to verify that what you have written works as you intend, I think this is especially important when it comes to system management code as you generally run it over a large number of nodes.

Personally I want to have confidence that I can update a `puppet module` and not break the entire infrastructure.

The _how_ is language/framework specific. For puppet modules this is generally done with `rspec` and the `rspec-puppet` plugin. There of puppet modules out there which are [really](https://github.com/puppetlabs/puppetlabs-apache) [good](https://github.com/puppetlabs/puppetlabs-vcsrepo) [examples](https://github.com/puppetlabs/puppetlabs-stdlib) of this. The general principles apply to different code too. e.g. [websites rendering](https://github.com/NERC-CEH/puppet-selenium), [javascript](http://jasmine.github.io/2.0/introduction.html) and [java](http://junit.org/junit4/)

**Pull request merged**